qiskit/pulse/schedule.py

```diff
@@ -1225,6 +1240,21 @@ def parameters(self) -> Set[Parameter]:
 
         return out_params
 
+    @property
+    def pulses(self):
+        """Get a list of all of the pulses used in this scheduleBlock."""
+        pulses = []
+        for instruction in self.instructions:
```

Comment:
Please directly iterate over `self.blocks` because calling `self.instructions` invokes typecast (lowering) to `Schedule`.